modules/internal/ChapelRange.chpl

```diff
@@ -144,7 +144,8 @@ module ChapelRange {
   
     var _low: idxType = 1;                         // lower bound
     var _high: idxType = 0;                        // upper bound
-    var _stride: strType = 1;                      // signed stride of range
+    var _stride: if stridable then strType else void =
+                   if stridable then 1 else _void; // signed stride of range
     var _alignment: idxType = 0;                   // alignment
     var _aligned : bool = false;
```

Comment:
This is a case I was discussing with Nikhil on email recently (and we talked about generating a warning for applying the align operator to an unstrided range indicating that it was a no-op, but I never implemented that).  What's happening is that the align operator normalizes the alignment to a value in 0..stride-1.  So if stride == 1, all indices are equivalent mod 1 and therefore we always store the alignment as 0 (so it's a no-op and we could warn against it... and ultimately should).